bigquery/bigquery_create_dataset_external/main.tf

```diff
@@ -0,0 +1,27 @@
+/**
+* Copyright 2024 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+# [START bigquery_create_dataset_external]
+resource "google_bigquery_dataset" "default" {
+  dataset_id    = "my_external_dataset"
+  friendly_name = "my_external_dataset"
```

Comment:
Maybe "my external dataset" to show that it doesn't need to conform to the same naming rule as `dataset_id`

Reply:
Done.